### PR TITLE
fix(RHTAPBUGS-560): update pipelines to use subdirectory

### PR DESCRIPTION
**Pull request by mmalina**
These changes will fix the issue of pipeline runs running in parallel overriding each other's data.

- create new pipeline versions
- use new version of collect-data with subdirectory
- use new version of apply-mapping that overrides original snapshot_spec file
- create new version of create-internal-requests that also uses a subdirectory and use it in the fbc-release pipeline
- make sure all pipelines clean up the subdirectory in the workspace at the end

### CODE REVIEW

**Comment by johnbieren** [comment]
Can you paste diffs or tell me what commit to look at? There are 13 commits in this pr

**Comment by mmalina** [comment]
> Can you paste diffs or tell me what commit to look at? There are 13 commits in this pr

You can skip all the "add new version" commits - they just add a duplicate of the old version. All the rest is relevant changes, broken into small chunks. I know it's a lot of commits, but I thought this would actually make the review easier - you see it bit by bit. OK, I squashed all the "add new version" commits into one. All the rest is the changes. I can squash further if you like. Let me know.

**Review by johnbieren — DISMISSED** [review]
hard to be sure with a change so big, but it lgtm

**Comment by mmalina** [comment]
> hard to be sure with a change so big, but it lgtm

thanks for the review. I intend to run the e2e against the new version(s) before merging. it will not cover everything, but at least some of it.

**Comment by mmalina** [comment]
I'm testing this with an e2e run here: https://github.com/redhat-appstudio/e2e-tests/pull/689

**Comment by mmalina** [comment]
One of the test runs in https://github.com/redhat-appstudio/e2e-tests/pull/689 revealed that I forgot something - I added a new commit. 

And now the latest test run seems to confirm that everything is ok: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-appstudio_e2e-tests/689/pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e/1696141129843478528

The test did fail, but all the release tests passed. It failed on rhtap-demo-suite with this error message:

> [FAIL] [rhtap-demo-suite] multi-component scenario with components without devfile or dockerfile [It] creates componentdetectionquery for component mc-withdockerfile-withoutdevfile [rhtap-demo]

This definitely doesn't seem anything release related.

So I will go ahead and merge this.

**Comment by mmalina** [comment]
Ah, it turns out that I'll need an approval again. @johnbieren can you take a look? See the new commit that I added yesterday.

**Comment by mmalina** [comment]
And the latest run in [the e2e pr](https://github.com/redhat-appstudio/e2e-tests/pull/689#issuecomment-1696964799https://github.com/redhat-appstudio/e2e-tests/pull/689#issuecomment-1696964799) passed! 🎉

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-appstudio_e2e-tests/689/pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e/1696437284498313216

**Pull request merged**
